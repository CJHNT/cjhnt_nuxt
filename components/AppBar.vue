<script setup>
const { loggedIn, user, clear, fetch } = useUserSession()
const router = useRouter()

fetch()
const logout = () => {
  clear()
  router.push({ name: 'index' })
}
</script>

<template>
  <v-app-bar color="#027f42">
    <template v-slot:prepend>
      <v-btn class="d-md-none"
        ><v-img
          src="@/assets/img/cjh-logo-quadratisch-klein.png"
          alt="Logo: CJHNT Project"
          width="32"
          inline
        ></v-img
      ></v-btn>
      <v-btn
        icon="mdi-information-outline"
        href="https://cjhnt-info.saw-leipzig.de/de"
        target="_blank"
      ></v-btn>
      <v-btn icon="mdi-home" href="/"></v-btn>
    </template>
    <template v-slot:append>
      <v-tooltip location="bottom" text="Advanced Search">
        <template v-slot:activator="{ props }">
          <v-btn icon="mdi-magnify" to="/search/advanced" v-bind="props"></v-btn>
        </template>
      </v-tooltip>

      <template v-if="loggedIn">
        <v-tooltip location="bottom" :text="`Logged in as ${user.email}`">
          <template v-slot:activator="{ props }">
            <v-btn icon="mdi-account" to="/auth/dashboard" v-bind="props"></v-btn>
            <v-btn icon="mdi-logout" @click="logout" v-bind="props"></v-btn>
          </template>
        </v-tooltip>
      </template>
      <v-btn v-else icon="mdi-login" to="/auth/login"></v-btn>
      <LangSwitcher />
    </template>
  </v-app-bar>
</template>
