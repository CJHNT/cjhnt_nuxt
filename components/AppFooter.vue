<script setup>
const props = defineProps({ border: { type: String, default: '' } })
</script>

<template>
  <v-footer class="d-flex flex-column" :app="true" :border="props.border">
    <v-container class="pa-0">
      <v-row no-gutters justify="center">
        <v-col cols="auto">
          <a href="https://www.akademienunion.de/forschungsprojekte/" target="_blank">
            <v-row>
              <v-col cols="auto">
                <picture>
                  <source
                    media="(max-width: 600px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="20"
                    width="60"
                  >
                  <source
                    media="(max-width: 960px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="30"
                    width="90"
                  >
                  <source
                    media="(max-width: 1280px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="40"
                    width="120"
                  >
                  <source
                    media="(max-width: 1920px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="50"
                    width="150"
                  >
                  <source
                    media="(max-width: 2560px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="60"
                    width="180"
                  >
                  <img
                    src="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    alt="Logo: Akademienunion"
                    height="70"
                    width="210"
                  >
                </picture>
              </v-col>
              <v-col cols="auto">
                <picture>
                  <source
                    media="(max-width: 600px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="20"
                    width="14"
                  >
                  <source
                    media="(max-width: 960px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="30"
                    width="21"
                  >
                  <source
                    media="(max-width: 1280px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="40"
                    width="28"
                  >
                  <source
                    media="(max-width: 1920px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="50"
                    width="35"
                  >
                  <source
                    media="(max-width: 2560px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="60"
                    width="42"
                  >
                  <img
                    src="@/assets/img/sachsen-signet_gruen.png"
                    alt="Logo: Akademienunion"
                    height="70"
                    width="47"
                  >
                </picture>
              </v-col>
              <v-tooltip activator="parent" location="top"
                >Dieses Forschungsvorhaben ist Teil im Akademienprogramm, das als derzeit größtes
                geistes- und kulturwissenschaftliches Langfrist-Forschungsprogramm der
                Bundesrepublik Deutschland von Bund und Ländern getragen wird. Diese Maßnahme wird
                mitfinanziert durch Steuermittel auf der Grundlage des vom Sächsischen Landtag
                beschlossenen Haushalts.</v-tooltip
              >
            </v-row>
          </a>
        </v-col>
      </v-row>
    </v-container>
    <div />
  </v-footer>
</template>

<style scoped>
.text-caption.footer-text {
  line-height: normal;
}
</style>

<style>
.v-tooltip__content {
  pointer-events: initial;
}
</style>
