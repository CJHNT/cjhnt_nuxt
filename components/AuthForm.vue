<script setup>
defineProps({
  loading: { type: Boolean, default: false },
  title: { type: String, required: true },
  label: { type: String, default: '' }
})
const emit = defineEmits(['submit'])
const email = ref('')
const repeatEmail = ref('')
const password = ref('')
const repeatPassword = ref('')
const showPassword = ref(false)
const showRepeatPassword = ref(false)
const wantsUpdates = ref(false)

function validateEmail() {
  const pattern =
    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  return pattern.test(email.value) || 'Invalid e-mail.'
}

const submit = () => {
  const payload = {
    email: email.value,
    repeatEmail: repeatEmail.value,
    password: password.value,
    repeatPassword: repeatPassword.value,
    wantsUpdates: wantsUpdates.value
  }
  emit('submit', payload)
}
</script>

<template>
  <v-card width="100%" class="mx-auto mt-5">
    <v-form @submit.prevent="submit">
      <v-card-title class="pb-0 text-h5">{{ $t(title) }}</v-card-title>
      <v-card-text>
        <v-text-field
          v-if="['auth.signUp', 'auth.changeEmail', 'auth.login'].includes(title)"
          v-model="email"
          :label="label ? $t(label) : 'Email'"
          :rules="[validateEmail()]"
          prepend-icon="mdi-email"
        />
        <v-text-field
          v-if="title === 'auth.changeEmail'"
          v-model="repeatEmail"
          :label="$t('auth.verifyEmail')"
          prepend-icon="mdi-email"
          :rules="[() => email === repeatEmail || $t('auth.noEmailMatch')]"
        />
        <v-text-field
          v-if="['auth.signUp', 'auth.changePassword', 'auth.login'].includes(title)"
          v-model="password"
          :type="showPassword ? 'text' : 'password'"
          :label="label ? $t(label) : $t('auth.password')"
          prepend-icon="mdi-lock"
          :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
          @click:append="showPassword = !showPassword"
        />
        <v-text-field
          v-if="['auth.signUp', 'auth.changePassword'].includes(title)"
          v-model="repeatPassword"
          :type="showRepeatPassword ? 'text' : 'password'"
          :label="$t('auth.verifyPassword')"
          prepend-icon="mdi-lock"
          :append-icon="showRepeatPassword ? 'mdi-eye' : 'mdi-eye-off'"
          :rules="[() => password === repeatPassword || $t('auth.noPasswordMatch')]"
          @click:append="showRepeatPassword = !showRepeatPassword"
        />
        <v-checkbox
          v-if="['auth.signUp'].includes(title)"
          v-model="wantsUpdates"
          :label="label ? $t(label) : $t('auth.sendMeUpdates')"
        />
      </v-card-text>
      <v-divider />
      <v-card-actions>
        <v-btn color="success" :disabled="loading" type="submit">
          <template v-if="loading">{{ $t('wait') }}</template>
          <template v-else-if="title === 'auth.changeUpdateStatus'">{{ $t(label) }}</template>
          <template v-else>{{ $t(title) }}</template>
        </v-btn>
      </v-card-actions>
      <template v-if="['auth.signUp', 'auth.login'].includes(title)">
        <v-divider />
        <v-card-text>
          <template v-if="title === 'auth.signUp'">
            {{ $t('auth.alreadyRegistered') }}
            <nuxt-link :to="{ name: 'auth-login' }">{{ $t('auth.login') }}</nuxt-link>
          </template>
          <template v-else-if="title === 'auth.login'">
            {{ $t('auth.noAccountYet') }}
            <nuxt-link :to="{ name: 'auth-register' }">{{ $t('auth.signUp') }}</nuxt-link>
          </template>
        </v-card-text>
      </template>
    </v-form>
  </v-card>
</template>
