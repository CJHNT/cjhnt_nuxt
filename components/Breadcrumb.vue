<script setup>
const props = defineProps({
  ancestors: { type: Array, required: true },
  index: { type: Number, required: true }
})
const { locale } = useI18n()
</script>

<template>
  <v-breadcrumbs :items="props.ancestors" class="py-0">
    <template v-slot:item="{ item }">
      <span v-if="item.disabled">{{ item.title[locale] }} {{ item.ref }}</span>
      <nuxt-link v-else :to="`/collection/${item.id}`">{{ item.title[locale] }}</nuxt-link>
    </template>
    <template v-slot:prepend>
      <v-icon v-if="index > 0" icon="mdi-subdirectory-arrow-right" size="small"></v-icon>
      <v-icon v-else icon="mdi-slash-forward" size="small"></v-icon>
    </template>
    <template v-slot:divider>
      <v-icon icon="mdi-slash-forward" size="small"></v-icon>
    </template>
  </v-breadcrumbs>
</template>
