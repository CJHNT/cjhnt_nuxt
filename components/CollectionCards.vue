<script setup>
const { locale } = useI18n()
const { data: theTexts } = await useAsyncData('apiAllTexts', async () => {
  const allTexts = await $fetch('http://127.0.0.1:5000/api/dts/collections', {
    query: { id: 'all_texts' }
  })
  const textPromises = allTexts.member.map(async (m) => {
    const { data: textData } = await useFetch('http://127.0.0.1:5000/api/dts/collections', {
      query: { id: m['@id'] }
    })
    const returnObject = {
      id: textData.value['@id'],
      de: textData.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'deu')[
        '@value'
      ],
      en: textData.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'eng')[
        '@value'
      ]
    }
    return returnObject
  })
  const finishedPromises = await Promise.all(textPromises)
  return finishedPromises
})
const sortedMembers = computed(() => {
  return [...theTexts.value].sort((a, b) => a.id.localeCompare(b.id))
})
import primaryImage from '@/assets/img/1_primary_texts.png'
import commentaryImage from '@/assets/img/commentary.jpg'

function getImage(urn) {
  if (urn === '1_primary_texts') {
    return primaryImage
  } else if (urn === 'urn:cts:cjhnt:commentary') {
    return commentaryImage
  }
}
</script>

<template>
  <v-row justify="center">
    <v-col v-for="member in sortedMembers" :key="member.id" cols="auto">
      <template>
        <nuxt-link class="mx-auto" :to="`/collection/${member.id}`" :title="member[locale]">
          <picture>
            <source media="(max-width: 600px)" :srcset="getImage(member.id)" height="100" />
            <source media="(max-width: 960px)" :srcset="getImage(member.id)" height="130" />
            <source media="(max-width: 1280px)" :srcset="getImage(member.id)" height="160" />
            <img :src="getImage(member.id)" :alt="member[locale]" height="200" />
          </picture>
        </nuxt-link>
      </template>
    </v-col>
  </v-row>
</template>
