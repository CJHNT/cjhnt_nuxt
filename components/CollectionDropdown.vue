<script setup>
defineProps({
  collId: {
    type: String,
    required: true,
    default: ''
  },
  collName: {
    type: String,
    required: true,
    default: ''
  }
})
</script>

<template>
  <div>
    <button>{{ collName }}</button>
    <p v-if="pending">Loading...</p>
    <CollectionDropdownContent v-else v-for="member in data.member" :key="member['@id']">{{
      member.title
    }}</CollectionDropdownContent>
  </div>
</template>
