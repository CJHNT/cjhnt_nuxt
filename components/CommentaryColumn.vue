<script setup>
import { allows } from 'nuxt-authorization/utils'
import { readClosed } from '~/utils/abilities'
import parentsAndSiblings from '~/utils/parentsAndSiblings'
import { getParentId, getEnDeBibliography, checkOpenText, getDocTitle } from '~/utils/getTextMeta'

const { locale } = useI18n()
const { user } = useUserSession()
const projectMember = await allows(readClosed, user.value)
const props = defineProps({
  urn: { type: String, default: '' },
  reff: { type: String, default: '' },
  index: { type: Number, default: 0 }
})
const allAncestors = defineModel({ type: Array, default: [] })
const ancestors = ref([])
const textMeta = await $fetch('/api/dts/collections', {
  body: { id: props.urn },
  method: 'POST'
})

const openText = checkOpenText(textMeta)
const docTitle = getDocTitle(textMeta)

const parentId = getParentId(textMeta) || props.urn.split('.').slice(0, -1).join('.')

const { textAncestors, collMembers } = await parentsAndSiblings(parentId)
ancestors.value = [...textAncestors, { id: props.urn, title: docTitle, disabled: true, ref: '' }]
allAncestors.value.push(ancestors.value)

const memberItems = collMembers
  .filter((m) => m['@id'] !== props.urn)
  .map((m) => {
    const docTitle = getDocTitle(m)
    const returnValue = {
      value: m['@id'],
      en: docTitle.en,
      de: docTitle.de,
      open: m['dts:dublincore']?.['dct:accessRights'] === 'open',
      props: { to: `/texts/${m['@id']}` }
    }
    return returnValue
  })
  .filter((c) => projectMember || c.open)

const citation = getEnDeBibliography(textMeta)

const engText = ref('')
const deuText = ref('')
const navReturn = ref(null)
const usedReff = ref(props.reff)
const validReffs = ref([])
const ntText = ref('')

if (openText || projectMember) {
  const navInfo = await $fetch('/api/dts/navigation', {
    body: { id: props.urn },
    method: 'POST'
  })
  navReturn.value = navInfo
  validReffs.value = navReturn.value['hydra:member'].map((r) => r.ref)
  if (!validReffs.value.includes(props.reff)) {
    usedReff.value = validReffs.value[0]
  }
  const formattedText = await $fetch('/api/dts/document', {
    body: { id: props.urn, ref: usedReff.value, xsl: 'assets/source/commentary.sef.json' },
    method: 'POST'
  })
  const parser = new DOMParser()
  const domText = parser.parseFromString(formattedText, 'text/html')
  const { data: apiNtText } = await useAsyncData('apiNtText', async () => {
    const ntElement = domText.getElementsByClassName('nt-source-text')[0]
    const ntSource = ntElement.getAttribute('source-text')
    const ntVerse = ntElement.getAttribute('source-verse')
    const apiResult = await $fetch('/api/dts/document', {
      body: { id: ntSource, ref: ntVerse, xsl: 'assets/source/nt_fragment.sef.json' },
      method: 'POST'
    })
    return apiResult
  })
  ntText.value = apiNtText.value
  engText.value = domText.getElementById('en-text')?.outerHTML ?? ''
  deuText.value = domText.getElementById('de-text').outerHTML
}
const engClass = computed(() => ({
  'h-0': locale.value === 'de',
  'overflow-hidden': locale.value === 'de'
}))
const deuClass = computed(() => ({
  'h-0': locale.value === 'en' && engText,
  'overflow-hidden': locale.value === 'en' && engText
}))
</script>

<template>
  <v-container>
    <v-row>
      <v-col cols="4" :order="index">
        <v-container>
          <v-row>
            <v-col>
              <!-- eslint-disable-next-line vue/no-v-html -->
              <div v-html="ntText" />
            </v-col>
          </v-row>
          <v-row class="my-4">
            <v-autocomplete
              :items="memberItems"
              :item-title="locale"
              density="compact"
              :hint="$t('comptext.readAnother')"
              width="200"
              persistent-hint
              clearable
            />
          </v-row>
          <v-row>
            <v-card>
              <v-card-title class="text-body-2 font-weight-bold">{{
                $t('comptext.suggestedCitation')
              }}</v-card-title>
              <v-card-text class="text-body-2">{{ citation[locale] }}</v-card-text>
            </v-card>
          </v-row>
        </v-container>
      </v-col>
      <v-col cols="8">
        <v-container>
          <v-row justify="center">
            <v-col cols="auto" class="pr-0">
              <h1>{{ docTitle[locale] }}</h1>
            </v-col>
            <v-col cols="auto" class="pl-0">
              <v-dialog max-width="500">
                <template #activator="{ props: activatorProps }">
                  <v-btn v-bind="activatorProps" variant="plain" size="x-small"
                    ><v-icon> mdi-copyright </v-icon></v-btn
                  >
                </template>

                <template #default="{ isActive }">
                  <v-card title="Copyright">
                    <v-card-text>{{ $t('comptext.commentaryLicense') }}</v-card-text>

                    <v-card-actions>
                      <v-spacer />

                      <v-btn :text="$t('closeDialog')" @click="isActive.value = false" />
                    </v-card-actions>
                  </v-card>
                </template>
              </v-dialog>
            </v-col>
          </v-row>
          <v-row class="text-content">
            <v-col>
              <!-- eslint-disable-next-line vue/no-v-html -->
              <div v-if="engText" :class="engClass" v-html="engText" />
              <!-- eslint-disable-next-line vue/no-v-html -->
              <div v-if="deuText" :class="deuClass" v-html="deuText" />
            </v-col>
          </v-row>
        </v-container>
      </v-col>
    </v-row>
  </v-container>
</template>

<style>
/* Style the tab */
.tab {
  overflow: hidden;
}

/* Style the buttons that are used to open the tab content */
.tab button {
  background-color: inherit;
  float: left;
  outline: none;
  cursor: pointer;
  padding: 1px;
  transition: 0.3s;
}

/* Change background color of buttons on hover */
.tab button:hover {
  background-color: #ddd;
}

/* Create an active/current tablink class */
.tab button.active {
  background-color: #ccc;
}

.sectionA p {
  font-size: smaller;
}

.beleg-section-name::after {
  content: ': ';
}

.beleg-section-name::before {
  content: '- ';
}
</style>
