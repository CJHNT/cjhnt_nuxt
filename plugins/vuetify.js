import '@/styles/main.scss'
export default defineNuxtPlugin(() => {})
