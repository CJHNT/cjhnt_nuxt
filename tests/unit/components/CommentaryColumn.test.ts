import { mountSuspended } from '@nuxt/test-utils/runtime'
import CommentaryColumn from '~/components/CommentaryColumn.vue'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'
import { createI18n } from 'vue-i18n'
import enLang from '~/lang/en.json'
import deLang from '~/lang/de.json'

const vuetify = createVuetify({
  components,
  directives
})

const i18n = createI18n({
  locale: 'en',
  messages: {
    en: enLang,
    de: deLang
  }
})

const i18nDe = createI18n({
  locale: 'de',
  messages: {
    en: enLang,
    de: deLang
  }
})

describe('Commentary Column Component', async () => {
  const enWrapper = await mountSuspended(CommentaryColumn, {
    props: { urn: 'urn:cts:cjhnt:commentary.1tim.001_001_002', reff: '1', index: 0 },
    global: {
      plugins: [vuetify, i18n]
    },
    attachTo: document.body
  })
  test('component mounts', () => {
    expect(enWrapper.html()).toMatchSnapshot()
  })
  test('switches correctly from en to de', async () => {
    const deWrapper = await mountSuspended(CommentaryColumn, {
      props: { urn: 'urn:cts:cjhnt:commentary.1tim.001_001_002', reff: '1', index: 0 },
      global: {
        plugins: [vuetify, i18nDe]
      }
    })
    expect(deWrapper.html()).toMatchSnapshot()
  })
  test('text without deu and eng tags on dc:title works', async () => {
    const wrapper = await mountSuspended(CommentaryColumn, {
      props: { urn: 'urn:cts:cjhnt:commentary.1tim.001_001_001', reff: '1', index: 0 },
      global: {
        plugins: [vuetify, i18nDe]
      }
    })
    expect(wrapper.find('h1').text()).toContain('1,1/1 Gott, der Retter: θεός σωτήρ')
  })
  test('call to document api with non-existent ref returns valid ref', async () => {
    const wrapper = await mountSuspended(CommentaryColumn, {
      props: { urn: 'urn:cts:cjhnt:commentary.1tim.001_001_001', reff: '2', index: 0 },
      global: {
        plugins: [vuetify, i18nDe]
      }
    })
    expect(wrapper.find('h1').text()).toContain('1,1/1 Gott, der Retter: θεός σωτήρ')
  })
  test('closed document is not shown', async () => {
    const wrapper = await mountSuspended(CommentaryColumn, {
      props: { urn: 'urn:cts:cjhnt:commentary.1tim.001_002_001', reff: '1', index: 0 },
      global: {
        plugins: [vuetify, i18nDe]
      }
    })
    expect(wrapper.find('#de-text').exists()).toBeFalsy()
  })
  // test('sibling menu is not empty', async () => {
  //   const clickElement = enWrapper.getComponent('[title=Open]')
  //   expect(clickElement.attributes('aria-label')).toBe('Open')
  //   await clickElement.trigger('click')
  //   expect(clickElement.attributes('aria-label')).toBe('Close')
  // })
  // I think this doesn't work because the textPage.js file is not loaded on the component but on the page
  // I will need to use an e2e test to get this to work
  // test('clicking belegstelle-erlautert reveals the text in section D', async () => {
  //   const clickElement = enWrapper.find('[class*=belegstelle-erl]')
  //   const targetEl = enWrapper.find(`${clickElement.attributes('data-target')}`)
  //   let expandElement = targetEl.element.parentElement
  //   expect(expandElement?.classList.contains('border-opacity-0')).toBe(true)
  //   await clickElement.trigger('click')
  //   expandElement = targetEl.element.parentElement
  //   expect(expandElement?.classList.contains('border-opacity-0')).toBe(false)
  // })
})
