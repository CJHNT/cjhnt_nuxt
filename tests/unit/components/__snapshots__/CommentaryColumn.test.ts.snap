// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Commentary Column Component > component mounts 1`] = `
"<div class="v-container v-locale--is-ltr">
  <div class="v-row">
    <div class="v-col v-col-4">
      <div class="v-container v-locale--is-ltr">
        <div class="v-row">
          <div class="v-col">
            <!-- eslint-disable-next-line vue/no-v-html -->
            <div></div>
          </div>
        </div>
        <div class="v-row my-4">
          <div class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-text-field v-autocomplete v-autocomplete--single" style="width: 200px;">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-owns="v-menu-2">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background" style="opacity: NaN;"></div>
                    <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!----><label class="v-label v-field-label" for="input-0">
                    <!---->
                    <!---->
                  </label>
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" value="">
                  </div>
                  <!---->
                </div>
                <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
                  <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear "></i></div>
                </transition-stub>
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-autocomplete__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <div class="v-input__details">
              <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
                <div class="v-messages__message">Read another text</div>
              </transition-group-stub>
              <!---->
            </div>
          </div>
        </div>
        <div class="v-row">
          <div class="v-card v-theme--light v-card--density-default v-card--variant-elevated">
            <!---->
            <div class="v-card__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="opacity: NaN;"></div>
                <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <!---->
            <div class="v-card-title text-body-2 font-weight-bold">Suggested Citation Form</div>
            <div class="v-card-text text-body-2">Herzer, Jens and Paulus Enke, "Commentary on 1 Timothy 1:1/2", in: Corpus Judaeo-Hellenisticum Novi Testamenti digital, [URL: https://cjhnt.saw-leipzig.de/texts/urn:cts:cjhnt:commentary.1tim.001_001_002]</div>
            <!---->
            <!----><span class="v-card__underlay"></span>
          </div>
        </div>
      </div>
    </div>
    <div class="v-col v-col-8">
      <div class="v-container v-locale--is-ltr">
        <div class="v-row justify-center">
          <div class="v-col v-col-auto pr-0">
            <h1>1,1/2 The Messiah as ἐλπίς</h1>
          </div>
          <div class="v-col v-col-auto pl-0"><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-plain" aria-haspopup="dialog" aria-expanded="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-copyright mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
            <!---->
          </div>
        </div>
        <div class="v-row text-content">
          <div class="v-col">
            <!-- eslint-disable-next-line vue/no-v-html -->
            <!--v-if-->
            <!-- eslint-disable-next-line vue/no-v-html -->
            <!--v-if-->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Commentary Column Component > switches correctly from en to de 1`] = `
"<div class="v-container v-locale--is-ltr">
  <div class="v-row">
    <div class="v-col v-col-4">
      <div class="v-container v-locale--is-ltr">
        <div class="v-row">
          <div class="v-col">
            <!-- eslint-disable-next-line vue/no-v-html -->
            <div></div>
          </div>
        </div>
        <div class="v-row my-4">
          <div class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-text-field v-autocomplete v-autocomplete--single" style="width: 200px;">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-owns="v-menu-2">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background" style="opacity: NaN;"></div>
                    <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!----><label class="v-label v-field-label" for="input-0">
                    <!---->
                    <!---->
                  </label>
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" value="">
                  </div>
                  <!---->
                </div>
                <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
                  <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear "></i></div>
                </transition-stub>
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-autocomplete__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <div class="v-input__details">
              <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
                <div class="v-messages__message">Einen anderen Text lesen</div>
              </transition-group-stub>
              <!---->
            </div>
          </div>
        </div>
        <div class="v-row">
          <div class="v-card v-theme--light v-card--density-default v-card--variant-elevated">
            <!---->
            <div class="v-card__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="opacity: NaN;"></div>
                <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <!---->
            <div class="v-card-title text-body-2 font-weight-bold">Empfohlene Zitierweise</div>
            <div class="v-card-text text-body-2">Herzer, Jens und Paulus Enke, "Kommentar zu 1. Timotheus 1,1/2", in: Corpus Judaeo-Hellenisticum Novi Testamenti digital, [URL: https://cjhnt.saw-leipzig.de/texts/urn:cts:cjhnt:commentary.1tim.001_001_002]</div>
            <!---->
            <!----><span class="v-card__underlay"></span>
          </div>
        </div>
      </div>
    </div>
    <div class="v-col v-col-8">
      <div class="v-container v-locale--is-ltr">
        <div class="v-row justify-center">
          <div class="v-col v-col-auto pr-0">
            <h1>1,1/2 Der Messias als ἐλπίς</h1>
          </div>
          <div class="v-col v-col-auto pl-0"><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-x-small v-btn--variant-plain" aria-haspopup="dialog" aria-expanded="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-copyright mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
            <!---->
          </div>
        </div>
        <div class="v-row text-content">
          <div class="v-col">
            <!-- eslint-disable-next-line vue/no-v-html -->
            <!--v-if-->
            <!-- eslint-disable-next-line vue/no-v-html -->
            <!--v-if-->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
